SwiftCompilerSources/Sources/Optimizer/ModulePasses/MandatoryPerformanceOptimizations.swift

```diff
@@ -293,6 +293,15 @@ fileprivate struct FunctionWorklist {
   }
 
   mutating func addAllPerformanceAnnotatedFunctions(of moduleContext: ModulePassContext) {
+    // For embedded Swift, optimize all the functions (there cannot be any
+    // generics, type metadata, etc.)
+    if moduleContext.options.enableEmbeddedSwift {
+      for f in moduleContext.functions {
```

Comment:
Also this loop does not belong to `addAllPerformanceAnnotatedFunctions`, because it has nothing to do with performance annotated functions. I suggest to move it to its own function.